instrumentation-core/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/httpclients/webclient/WebClientBeanPostProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.httpclients.webclient;
+
+import io.opentelemetry.instrumentation.springwebflux.client.WebClientTracingFilter;
+import io.opentelemetry.trace.Tracer;
+import java.util.List;
+import java.util.function.Consumer;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * Inspired by: <a
+ * href="https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientAutoConfiguration.java">Spring
+ * Cloud Sleuth</a>
+ */
+final class WebClientBeanPostProcessor implements BeanPostProcessor {
+
+  private final Tracer tracer;
+
+  public WebClientBeanPostProcessor(Tracer tracer) {
+    this.tracer = tracer;
+  }
+
+  @Override
```

Comment:
[Question] Do you need to override this, since it's the default behaviour?

Reply:
Actually this is the default behavior of postProcessBeforeInitialization. The default implementation of this method is to return the bean object and that's what I do here. This override is a leftover from when I was experimenting with java 7. Since java 8 has default implementations for interfaces this is now unnecessary. I'll remove this override in my next commit.